charts/dns/values.yaml

```
@@ -30,17 +107,77 @@ externalDNS:
       cpu: 1000m
       memory: 256Mi
 
+  sources:
+    - ambassador-host
+    - contour-httpproxy
+    - contour-ingressroute
+    - crd
+    - gloo-proxy
+    - ingress
+    - istio-gateway
+    - istio-virtualservice
+    - kong-tcpingress
+    - node
+    - openshift-route
+    - pod
+    - service
+    - skipper-routegroup
+
+  logLevel: info  # panic, debug, info, warning, error, fatal
+  logFormat: text  # text, json
+  interval: 30s
+  triggerLoopOnEvent: true
+  policy: upsert-only  # sync, upsert-only.
+  txtOwnerId: ""
+  txtPrefix: ""
+  txtWildcardReplacement: ""
+  managedRecordTypes: A,CNAME,NS
+
 pdns:
+  nameOverride: ""
+  fullnameOverride: ""
+
   registry: ghcr.io
   repository: txqueuelen/pdns-stateless
   tag: v4.4
   pullPolicy: IfNotPresent
 
-  # Allows to override the name of the objects related to PowerDNS
-  nameOverride: ""
+  securityContext:
+    allowPrivilegeEscalation: false
+    capabilities:
+      add:
+      - NET_BIND_SERVICE
+      drop:
+      - all
+    # readOnlyRootFilesystem: true
+    # runAsNonRoot: true
+    # runAsUser: 1000
+
+  resources:
+    requests:
+      cpu: 50m
+      memory: 128Mi
+    limits:
+      cpu: 1000m
+      memory: 256Mi
+
+  apiKeySecret:
+    # Specifies whether the secret should be created
+    create: true
+    # Annotations to add to the secret
+    annotations: {}
+    # The name of the secret to use.
+    # If not set and create is true, a name is generated using the fullname template
+    # If set and create is false, user should create the secret by himself and this is the name used.
+    name: ""
```

This seems like a large amount of thinking I need to do, while the chart could be doing it instead.

```suggestion
    # The name of the secret to use.
    # If not set and create is true, a name is generated using the fullname template
    # If set and create is false, user should create the secret by themselves and this is the name used.
    name: ""
```

---

I will rephrase it...

---

I like the pattern we've followed other times when you have an `existingSecretName`. If it is empty, the chart understands it has to create the secret. If it isn't, the chart will reference that secret instead.

---

I have not committed the change yet. The new phrase is like this:
> The name of the secret to use.
> If not set and create is true, a name is generated using the fullname template
> You can set your own secret by setting `create` to false and point this `name` to your secret's name.

---

Why make me handle two variables, where some combinations of them are unreasonable, instead of just one?

---

`name` is for the name of the secret and `key` for the key of the secret.

If you do not set nothing you don't have to set them. you have only to set `valueBase64` or `valuePlain` and the chart makes the rest.

But you can set your own secret and key where to look for the API Key.

---

> If not set and create is true

By two variables I mean this, one variable for the name (which only makes sense if the secret is external) and another one to tell whether to create the secret or not. My suggestion is that `create` is `name != ""`

---

Oh! This is a common pattern. In the same way that we have an enable for services and ingresses.

In fact this has the same exact patter as a `ServiceAccount`. The name used depends on if it is enabled or not. And the name changes in case you have provided one or not:
https://github.com/txqueuelen/pdns-stateless/pull/18/files#diff-19d4c48f49b8fa389619de9b21d45d673e47521da8acbf24e1bf84915aa2ca93R51-R57

In case you create it, the name is used to point to a secret you should provide.

I don't like the magic of `""` means something and `"something"` do something completely different just because it has a value...

---

Being able to specify the name for the secret/sac that the chart will create seems a bit pointless to me... But it's true that it's a common pattern, I did not realize that. Let's go for it.